sdk/communication/Azure.Communication.CallAutomation/tests/CallConnections/CallConnectionAutomatedLiveTests.cs

```diff
@@ -101,5 +101,86 @@ public async Task RemoveAUserCallTest()
+        [RecordedTest]
+        public async Task CancelAddParticipantTest()
+        {
+            // create caller and receiver
+            var user = await CreateIdentityUserAsync().ConfigureAwait(false);
+            var target = await CreateIdentityUserAsync().ConfigureAwait(false);
+            var participantToAdd = await CreateIdentityUserAsync().ConfigureAwait(false);
+            var client = CreateInstrumentedCallAutomationClientWithConnectionString(user);
+            var targetClient = CreateInstrumentedCallAutomationClientWithConnectionString(target);
+            string? callConnectionId = null;
+
+            try
+            {
+                // setup service bus
+                var uniqueId = await ServiceBusWithNewCall(user, target);
+
+                // create call and assert response
+                var createCallOptions = new CreateCallOptions(new CallInvite(target), new Uri(TestEnvironment.DispatcherCallback + $"?q={uniqueId}"));
+                CreateCallResult response = await client.CreateCallAsync(createCallOptions).ConfigureAwait(false);
+                callConnectionId = response.CallConnectionProperties.CallConnectionId;
+                Assert.IsNotEmpty(response.CallConnectionProperties.CallConnectionId);
+
+                // wait for incomingcall context
+                string? incomingCallContext = await WaitForIncomingCallContext(uniqueId, TimeSpan.FromSeconds(20));
+                Assert.IsNotNull(incomingCallContext);
+
+                // answer the call
+                var answerCallOptions = new AnswerCallOptions(incomingCallContext, new Uri(TestEnvironment.DispatcherCallback));
+                AnswerCallResult answerResponse = await targetClient.AnswerCallAsync(answerCallOptions);
+
+                // wait for callConnected
+                var connectedEvent = await WaitForEvent<CallConnected>(callConnectionId, TimeSpan.FromSeconds(20));
+                Assert.IsNotNull(connectedEvent);
+                Assert.IsTrue(connectedEvent is CallConnected);
+                Assert.AreEqual(callConnectionId, ((CallConnected)connectedEvent!).CallConnectionId);
+
+                // add participant
+                var callConnection = response.CallConnection;
+                var operationContext = "context";
+                var addParticipantOptions = new AddParticipantOptions(new CallInvite(participantToAdd))
+                {
+                    InvitationTimeoutInSeconds = 60,
+                    OperationContext = operationContext,
+                };
+                var addParticipantResponse = await callConnection.AddParticipantAsync(addParticipantOptions);
+                Assert.AreEqual(operationContext, addParticipantResponse.Value.OperationContext);
+                Assert.IsNotNull(addParticipantResponse.Value.InvitationId);
+
+                // ensure invitation has arrived
+                await Task.Delay(3000);
```

Comment:
Do you think we can wait for the incoming call instead?

Reply:
I think it's fine, WaitForIncomingCallContext is also timer based method anyway so the issue of flakiness would still be there.